### DISCUSSION TITLE
HA Shelly pro 3EM Electricity

### DISCUSSION THREAD

**Discussion by G0thm** [Arnold-n/P1P2MQTT#131] [Q&A]
How can I add the power consumption for P1P2 to Home Assistant? 
I have a Shelly Pro 3em that shows me the power consumption through the Shelly integration.
I just can't get the mqtt to run additionally.

Can anyone help me with this?

**Comment by jirkaherzig** [comment]
Hi,
I have got this integrated, not sure if correctly, but this is the way:
in shelly PRO3EM console, in Scripts, I have a script like this: 
```
// set this to >0 then log is activ
let log = 0;

// set this to false to stop publishing on MQTT
let MQTTpublish = true;

//published variables
let lastPublishedMQTT_Total_Act_Power = "";


// query the MQTT prefix on startup
let SHELLY_ID = undefined;

///
Shelly.call("Mqtt.GetConfig", "", function(res, err_code, err_msg, ud) {
    SHELLY_ID = res["topic_prefix"];
});

///
function timerHandler(user_data) {
    let em = Shelly.getComponentStatus("em", 0);
    if (typeof em.total_act_power !== 'undefined') {

        // a act power
        let value = em.total_act_power.toFixed(1);
        if (value !== lastPublishedMQTT_Total_Act_Power) {
            MQTT.publish(
                "P1P2/P/meter/U/9/Electricity_Power",
                value,
                0,
                false
            );
            lastPublishedMQTT_Total_Act_Power = value;
            if (log > 0)
                print("Total_Act_Power: " + value);
        }       
    }
}

Timer.set(3000, true, timerHandler, null);
```

this will publish energy consumption to your MQTT broker to P1P2 topic. 

For me, the "HC_Power Consumption_Heatpump" is changing over time. 
Unfortunately, no P1P2 meters are working for me. 

Maybe someone else will help. 
Jirka

**Comment by Arnold-n** [comment]
Thanks for your script!

Does your system have any counters which are visible via Daikin's wall room thermostat? If so, a log during a manual action when entering that menu could be helpful.

**Comment by Thirstysumo** [comment]
let log = 1;
let MQTTpublish = true;

let SHELLY_ID;
Shelly.call("Mqtt.GetConfig", "", function (res) {
  SHELLY_ID = res["topic_prefix"];
  print("MQTT prefix: " + SHELLY_ID);
});

function publishPower() {
  let em = Shelly.getComponentStatus("em1", 0);

  if (em && typeof em.act_power === 'number') {
    let value = em.act_power.toFixed(1);

    if (MQTTpublish) {
      MQTT.publish("P1P2/P/meter/U/9/Electricity_Power", value, 0, false);
    }

    if (log > 0) {
      print("Published: " + value + " W");
    }
  } else {
    print("Power reading unavailable");
  }
}

// Publish every 60 seconds no matter what
Timer.set(60000, true, publishPower);

**Comment by Thirstysumo** [comment]
This worked for me, remember to make sure you get the correct Shelly device, em1 or em
